DRILL-7054: PCAP timestamp in milliseconds

jculio opened the pull request:
A column has been added for retrieving the timestamp in microseconds. 
The pcap provides microseconds accuracy and such a parameter is really important for getting statistics from packet captures. In the timestamp column, instead, it is exported as DATE format which has millisecond accuracy.
I could not compute many of the feature I had to work on before this patch.

priteshm commented:
@arina-ielchiieva @vdiravka could you please review?

cgivre commented:
I can do review if you'd like.  This looks good, but can we add a unit test?  Also, we should probably do this for PCAP-NG if it doesn't already, but that's a separate issue.

jculio commented:
How can I add the unit test without doing again the pull request?

vdiravka commented:
> How can I add the unit test without doing again the pull request?

Thank you for the PR. Just update your local branch with a new changes and push it your remote repo.
_Note_: please squash your commits into one (here is no need to have both of them).

cgivre commented:
HI @jculio 
Take a look at: https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/exec/store/pcap/TestPcapDecoder.java#L57. 

Specifically here:  
https://github.com/apache/drill/blob/fee4b1cdc1a1bf4ac3dfdbbf435a5696540ce59a/exec/java-exec/src/test/java/org/apache/drill/exec/store/pcap/TestPcapDecoder.java#L57

You can just add one line to call your new function and then verify the result.  Once you've done that, run the test locally to make sure that it works:
`mvn =Dtest=TestPcapDecoder#testBasics test` and assuming it does, push it to your remote repo.  

As @vdiravka said, you'll need to squash your commits into one.  Here's how: After verifying that everything is working, count how many commits you have made.
Then:

```
git reset --soft HEAD~3
git commit -m "DRILL-7054: timestamp in milliseconds"
git push origin +master
```

The first line there uncommits the last `n` commits you made.  The second line then recommits everything into one commit.  Note that Drill requires commits to have a message identifying the PR.  The final line pushes everything back to github and overwrites the previous commits on github with one commit.  

Thank you very much for this contribution!!

jculio commented:
I tied to follow your instruction by doing:

- I created the same file
https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/exec/store/pcap/TestPcapDecoder.java
<https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/exec/store/pcap/TestPcapDecoder.java#L57>

- I added at line 57 the code:

@Test

public void testMicrotQuery() throws Exception {

  runSQLVerifyCount("select `timestamp_micro` from
dfs.`store/pcap/cap-test.pcap`", 1);

}

where cap-test.pcap is the file I added in the pcap folder.


-Then I run the command

mvn -Dtest=TestPcapDecoder#testMicrotQuery test and I got the error:


Failed to execute goal
org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M2:test

Failed to execute goal
org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M2:test (default-test)
on project drill-protocol: No tests were executed!


First time running/writing unit test, so I will need help.

Thank you very much




Il giorno ven 1 mar 2019 alle ore 15:27 Charles S. Givre <
notifications@github.com> ha scritto:

> HI @jculio <https://github.com/jculio>
> Take a look at:
> https://github.com/apache/drill/blob/master/exec/java-exec/src/test/java/org/apache/drill/exec/store/pcap/TestPcapDecoder.java#L57
> .
>
> Specifically here:
>
> https://github.com/apache/drill/blob/fee4b1cdc1a1bf4ac3dfdbbf435a5696540ce59a/exec/java-exec/src/test/java/org/apache/drill/exec/store/pcap/TestPcapDecoder.java#L57
>
> You can just add one line to call your new function and then verify the
> result. Once you've done that, run the test locally to make sure that it
> works:
> mvn =Dtest=TestPcapDecoder#testBasics test and assuming it does, push it
> to your remote repo.
>
> As @vdiravka <https://github.com/vdiravka> said, you'll need to squash
> your commits into one. Here's how: After verifying that everything is
> working, count how many commits you have made.
> Then:
>
> git reset --soft HEAD~3
> git commit -m "DRILL-7054: timestamp in milliseconds"
> git push origin +master
>
> The first line there uncommits the last n commits you made. The second
> line then recommits everything into one commit. Note that Drill requires
> commits to have a message identifying the PR. The final line pushes
> everything back to github and overwrites the previous commits on github
> with one commit.
>
> Thank you very much for this contribution!!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/drill/pull/1665#issuecomment-468681971>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/At31l2Qek0NUqNA4H8HUZnPXxf7-orlEks5vSTjEgaJpZM4bXgAB>
> .
>

cgivre commented:
Hi @jculio 
You acutally don't have to create a new file.   Just add a line to the `TestPcapDecoder`.

All you have to do is add:

```
assertEquals("<value>", p. getTimestampMicro());
```
and replace the `<value>` with the expected value.

jculio commented:
Hi @Charles Givre <cgivre@gmail.com> , I followed your instruction and I
pushed :)

Il giorno ven 1 mar 2019 alle ore 20:36 Charles S. Givre <
notifications@github.com> ha scritto:

> Hi @jculio <https://github.com/jculio>
> You acutally don't have to create a new file. Just add a line to the
> TestPcapDecoder.
>
> All you have to do is add:
>
> assertEquals("<value>", p. getTimestampMicro());
>
> and replace the <value> with the expected value.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/drill/pull/1665#issuecomment-468785456>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/At31l9RxQ_M-2lHdOYAc8TTxjFJTMaFDks5vSYEqgaJpZM4bXgAB>
> .
>

cgivre commented:
LGTM +1
